src/particles.cpp

```diff
@@ -431,31 +428,42 @@ ParticleManager::~ParticleManager()
 
 void ParticleManager::step(float dtime)
 {
-	stepParticles (dtime);
-	stepSpawners (dtime);
+	stepParticles(dtime);
+	stepSpawners(dtime);
+	stepSpawnersLocal(dtime);
 }
 
 void ParticleManager::stepSpawners (float dtime)
 {
 	MutexAutoLock lock(m_spawner_list_lock);
 	for (std::map<u32, ParticleSpawner*>::iterator i =
 			m_particle_spawners.begin();
-			i != m_particle_spawners.end();)
-	{
-		if (i->second->get_expired())
-		{
+			i != m_particle_spawners.end(); i++) {
+		if (i->second->get_expired()) {
 			delete i->second;
-			m_particle_spawners.erase(i++);
+			m_particle_spawners.erase(i);
```

Comment:
are you sure this works?
erasing & iterating at the same time needs some special care